cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/AdaptTemplatesCommand.java

```diff
@@ -0,0 +1,239 @@
+        String classesJarPath = cobigenUtils.getTemplatesJar(false).getPath().toString();
+        FileSystem fileSystem = FileSystems.getDefault();
+
+        if (destinationPath == null) {
+            throw new IOException("Cobigen folder path not found!");
+        }
+
+        String templatesFolderPath =
+            fileSystem.getPath(destinationPath + File.separator + ConfigurationUtils.COBIGEN_TEMPLATES).toString();
+
+        if (templatesFolderPath == null) {
+            throw new IOException("Cobigen templates folder path not found!");
+        }
+
+        Path cobigenTemplatesFolderPath = null;
+        if (fileSystem != null && destinationPath != null) {
+            cobigenTemplatesFolderPath = fileSystem.getPath(templatesFolderPath);
+        }
+
+        if (cobigenTemplatesFolderPath == null) {
+            throw new IOException(
+                "An exception occurred while processing Jar files to create CobiGen_Templates folder");
+        }
+
+        logger.debug("Processing jar file @ {}", sourcesJarPath);
+
+        // If we are unzipping a sources jar, we need to get the pom.xml from the normal jar
+        if (sourcesJarPath.contains("sources")) {
+            try (ZipFile file = new ZipFile(classesJarPath)) {
+                Enumeration<? extends ZipEntry> entries = file.entries();
+                if (Files.notExists(cobigenTemplatesFolderPath)) {
+                    Files.createDirectory(cobigenTemplatesFolderPath);
+                }
+                while (entries.hasMoreElements()) {
+                    ZipEntry entry = entries.nextElement();
+                    if (entry.getName().equals("pom.xml")) {
+                        Path saveForFileCreationPath =
+                            fileSystem.getPath(templatesFolderPath + File.separator + entry.getName());
+                        createFile(file, entry, saveForFileCreationPath);
+                    }
+                }
+            } catch (IOException e) {
+                throw new IOException(
+                    "An exception occurred while unpacking pom.xml from Jar file to templates folder");
+            }
+        }
+
+        // unpack sources
+        try (ZipFile file = new ZipFile(sourcesJarPath)) {
+            Enumeration<? extends ZipEntry> entries = file.entries();
+            if (Files.notExists(cobigenTemplatesFolderPath)) {
+                Files.createDirectory(cobigenTemplatesFolderPath);
+            }
+            while (entries.hasMoreElements()) {
+                ZipEntry entry = entries.nextElement();
+                Path saveForFileCreationPath =
+                    fileSystem.getPath(templatesFolderPath + File.separator + entry.getName());
+                if (entry.getName().contains("context.xml")) {
+                    saveForFileCreationPath =
+                        fileSystem.getPath(templatesFolderPath + File.separator + entry.getName());
+                } else if (entry.getName().contains("com/")) {
```

Comment:
com? this sounds strange. It's a package, which could be anything, why do you check on com/?